zebra-rpc/src/methods.rs

```diff
@@ -362,3 +459,44 @@ pub struct GetBlock(#[serde(with = "hex")] SerializedBlock);
 ///
 /// Also see the notes for the [`Rpc::get_best_block_hash` method].
 pub struct GetBestBlockHash(#[serde(with = "hex")] block::Hash);
+
+/// Response to a `getrawtransaction` RPC request.
+///
+/// See the notes for the [`Rpc::get_raw_transaction` method].
+// pub struct GetRawTransaction(pub(super) SerializedTransaction);
+#[derive(Clone, Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+pub enum GetRawTransaction {
+    /// The raw transaction, encoded as hex bytes.
+    Raw(#[serde(with = "hex")] Vec<u8>),
+    /// The transaction object.
+    Object {
+        /// The raw transaction, encoded as hex bytes.
+        #[serde(with = "hex")]
+        hex: Vec<u8>,
+        /// The height of the block that contains the transaction, or -1 if
+        /// not applicable.
+        height: i32,
+    },
+}
+
+impl GetRawTransaction {
+    fn from_transaction(
+        tx: Arc<Transaction>,
+        height: Option<block::Height>,
+        verbose: bool,
+    ) -> std::result::Result<Self, io::Error> {
+        let serialized_tx = tx.zcash_serialize_to_vec()?;
+        if verbose {
+            Ok(GetRawTransaction::Object {
+                hex: serialized_tx,
+                height: match height {
+                    Some(height) => height.0 as i32,
+                    None => -1,
```

Comment:
How do we know lightwalletd is expecting a `-1` for when transactions are not yet in a block ? Can we confirm this behavior with a link to the lightwalletd code ? Thanks!

Reply:
https://github.com/zcash/lightwalletd/blob/83bb19918fb36a5f6d5013fba7938daf3d73db1d/walletrpc/service.proto#L39

Reply:
I missed the lightwalletd handling, I had to dig into `zcash` source code: https://github.com/zcash/zcash/blob/96d6ee0b8f76f3f4587b5080dbf7cdf8683e0963/src/rpc/rawtransaction.cpp#L260-L268